test/Microsoft.AspNet.Routing.Tests/RouteCollectionTest.cs

```diff
@@ -49,7 +49,38 @@ public class RouteCollectionTest
             Assert.Same(target.Object, pathData.Router);
             Assert.Empty(pathData.DataTokens);
         }
-        
+
+        [Theory]
+        [InlineData(@"Home/Index/23", "/home/index/23/", true)]
+        [InlineData(@"Home/Index/23", "/Home/Index/23", false)]
+        [InlineData(@"Home/Index/23/?Param1=ABC&Param2=Xyz", "/Home/Index/23/?Param1=ABC&Param2=Xyz", true)]
+        [InlineData(@"Home/Index/23?Param1=ABC&Param2=Xyz", "/Home/Index/23/?Param1=ABC&Param2=Xyz", true)]
+        [InlineData(@"Home/Index/23#Param1=ABC&Param2=Xyz", "/Home/Index/23#Param1=ABC&Param2=Xyz", false)]
+        [InlineData(@"Home/Index/23#Param1=ABC&Param2=Xyz", "/Home/Index/23/#Param1=ABC&Param2=Xyz", true)]
+        public void GetVirtualPath_CanAppendTrailingSlash_BasedOnOptions(
```

Comment:
I'd personally write all of these tests as `appendTrailingSlash = true`. We have lots of tests already that **don't** add a URL.

I'd be more interested in some tests that mix lowercase with this feature since it's all in one big method.